src/backend/distributed/planner/multi_planner.c

```diff
@@ -117,6 +117,8 @@ multi_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 
 		if (needsDistributedPlanning)
 		{
+			relationRestrictionContext->parseTree = copyObject(originalQuery);
```

Comment:
I think we should move this line to `CreateAndPushPlannerContexts ()` to make it more readable